manager/scheduler/filter.go

```diff
@@ -61,13 +61,34 @@ func (f *ResourceFilter) SetTask(t *api.Task) bool {
 	if r == nil || r.Reservations == nil {
 		return false
 	}
-	if r.Reservations.NanoCPUs == 0 && r.Reservations.MemoryBytes == 0 {
+
+	res := r.Reservations
+	if res.NanoCPUs == 0 && res.MemoryBytes == 0 && len(res.ThirdParty) == 0 {
 		return false
 	}
+
 	f.reservations = r.Reservations
 	return true
 }
 
+// HasEnoughThirdPartyResources returns true if node can satisfy task TPR's request
+func HasEnoughThirdPartyResources(nodeRes *api.ThirdPartyResource, taskRes *api.ThirdPartyResource) bool {
```

Comment:
I'd suggest moving this and `ClaimThirdPartyResource` to an `api/thirdpartyresource` package (or whatever package name is appropriate given the final naming decisions). Then we can avoid constraintenforcer importing the scheduler code.